src/app/core/guards/current-user.ts

```diff
@@ -76,32 +93,38 @@ currentSubscription: Subscription=null;
 
 
    }
-
-   public async defaultBusiness() {
-    await this.database.get(PouchConfig.Tables.business).then(res=> {
-
-        const currentBusiness: Business=res.businesses.find(business=>business.active);
-        this.eventBus.publish(new CurrentBusinessEvent(currentBusiness));
-
-    },error=> {
-        if(error.error && error.status==='404' ||  error.status===404) {
-          this.eventBus.publish(new CurrentBusinessEvent(null));
-        }
-    });
-
+   
+   public async defaultBusiness(userId) {
+
+   
+        await this.database.activeBusiness(userId).then(res=>{
+
+            if(res.docs && res.docs.length > 0){
+                    this.eventBus.publish(new CurrentBusinessEvent(res.docs[0]));
+                    this.defaultBranch();
+            }
+
+        },error=> {
+            if(error.error && error.status==='404' ||  error.status===404) {
+              this.eventBus.publish(new CurrentBusinessEvent(null));
+            }
+        });
+     
   }
 
 
   public async defaultBranch() {
-    await this.database.get(PouchConfig.Tables.branches).then(res=> {
-      const currentBranch: Branch=res.branches.find(branch=>branch.active);
-      this.eventBus.publish(new CurrentBranchEvent(currentBranch));
```

Comment:
I have the same issue here, the current branch I am worried that we can get a null response from it.